python/triton/experimental/gluon/language/_semantic.py

```diff
@@ -155,6 +155,8 @@ def shared_load(self, mem_desc, layout):
         return ttgl.tensor(handle, ret_ty)
 
     def shared_store(self, mem_desc, value):
+        assert value.shape == mem_desc.shape, f"source shape {value.shape} and destination shape {mem_desc.shape} must match"
+        assert value.dtype == mem_desc.dtype, f"source dtype {value.dtype} and destination dtype {mem_desc.dtype} must match"
```

Comment:
Looks like this is already validated in the IR?

Reply:
It is, but the IR errors are harder to read than errors from the frontend and they don't generate a `CompilerError` with the full call context.

Reply:
In terms of re-using the IR error, we could parse the MLIR diagnostics and regenerate the full call trace, but because the verifier error happens after the frontend run, there will be less information in the error. This would potentially require adding "called from" information into the MLIR locations so that they can be recovered from the IR as well.

Reply:
Can we just call verify for the specific ops we're building as we go?